enterprise/server/remote_execution/containers/docker/docker.go

```diff
@@ -160,50 +173,36 @@ func (r *dockerCommandContainer) Run(ctx context.Context, command *repb.Command,
 		}()
 	}()
 
-	r.copyContainerLogs(ctx, cid, result)
-	if result.Error != nil {
+	var stdout, stderr bytes.Buffer
+	eg := &errgroup.Group{}
+	eg.Go(func() error {
+		_, errCopy := stdcopy.StdCopy(&stdout, &stderr, hijackedResp.Reader)
+		return wrapDockerErr(errCopy, "failed to copy docker container output")
+	})
+	eg.Go(func() error {
+		statusCh, errCh := r.client.ContainerWait(ctx, cid, dockercontainer.WaitConditionNotRunning)
+		select {
+		case err := <-errCh:
+			return wrapDockerErr(err, "container did not exit cleanly")
+		case s := <-statusCh:
+			containerTerminated = true
+			if s.Error != nil {
+				return wrapDockerErr(status.UnknownError(s.Error.Message), "failed to get container status")
+			}
+			result.ExitCode = int(s.StatusCode)
+			return nil
+		}
+	})
+	if err := eg.Wait(); err != nil {
+		result.Error = wrapDockerErr(err, "failed to run docker container")
 		return result
 	}
 
-	// The container should not be running at this point since we're done copying
-	// logs, but we need to call ContainerWait to get the exit code.
-	statusCh, errCh := r.client.ContainerWait(ctx, cid, dockercontainer.WaitConditionNotRunning)
-
-	select {
-	case err := <-errCh:
-		result.Error = wrapDockerErr(err, "container did not exit cleanly")
-	case s := <-statusCh:
-		exitedCleanly = true
-		if s.Error != nil {
-			result.Error = wrapDockerErr(status.UnknownError(s.Error.Message), "failed to get container status")
-			return result
-		}
-		result.ExitCode = int(s.StatusCode)
-	}
+	result.Stdout = stdout.Bytes()
+	result.Stderr = stderr.Bytes()
```

Comment:
we want to run this code if `err != nil` as well, to get partial output (useful for debugging).